### PR TITLE
[#117] Add LLM locally

### LINKED ISSUE #117: Add LLM locally
Currently we create requests to LLM APIs for local development. The purpose of this is to find a solution that doesn't involve making calls to LLM APIs

### PR DESCRIPTION
**Pull request by JdFSilva**
Add makefile commands to run ollama with llama3.2 locally
Add class to bypass liteLLM and connect directly to local llama3.2
Add env variable specifically for handling local LLMs

Closes #117

### CODE REVIEW

**Review by jfrverdasca — APPROVED** [review]

File: labs/config/settings.py [hunk 1]
```diff
@@ -66,3 +66,5 @@
 
 REDIS_HOST = os.environ.get("REDIS_HOST")
 REDIS_PORT = os.environ.get("REDIS_PORT")
+
+LOCAL_LLM = eval(os.environ.get("TEST_ENVIRONMENT", "False"))
```
Comment (kenvontucky) [inline labs/config/settings.py:70]: shouldn't this be LOCAL_LLM?

File: tests/test_local_llm.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+from unittest.mock import patch
+
+from labs.llm import call_llm_with_context
+
+
+class TestLocalLLM:
+    @patch("labs.llm.vectorize_to_database")
+    @patch("labs.llm.find_similar_embeddings")
+    def test_local_llm(self, mocked_context, mocked_vectorize_to_database):
```
Comment (kenvontucky) [inline tests/test_local_llm.py:9]: lets set the LOCAL_LLM env here to make things clearer

**Review by kenvontucky — CHANGES_REQUESTED** [review]

File: labs/config/settings.py [hunk 3]
```diff
@@ -66,3 +66,5 @@
 
 REDIS_HOST = os.environ.get("REDIS_HOST")
 REDIS_PORT = os.environ.get("REDIS_PORT")
+
+LOCAL_LLM = eval(os.environ.get("LOCAL_LLM", "False"))
```
Comment (jpb18) [inline labs/config/settings.py:70]: Newbie question: why do we need the eval here?

File: labs/litellm_service/local.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
+from ollama import Client
+
+
+class RequestLocalLLM:
+    def __init__(self):
+        self.client = Client(host="http://localhost:11434")
```
Comment (jpb18) [inline labs/litellm_service/local.py:6]: Should this be hardcoded?

File: labs/litellm_service/local.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+        [
+            {
+                "role": "user",
+                "content": ""
+            }
+        ]
+        Where role can be "user", "assistant", "system".
+        And content is the body of the message.
+        """
+        model = "llama3.2"
```
Comment (jpb18) [inline labs/litellm_service/local.py:23]: Any reason why this is a constant?

File: labs/litellm_service/request.py [hunk 6]
```diff
@@ -27,31 +26,6 @@ class RequestLiteLLM:
...
-        }
-        data = {"messages": messages}
-        result = requests.post(
-            f"http://0.0.0.0:4000/chat/completions?model={model}",
-            headers=headers,
-            json=data,
-        )
-
-        return result.json()
-
```
Comment (jpb18) [inline labs/litellm_service/request.py:54]: Why was this removed?

**Review by jpb18 — APPROVED** [review]

File: labs/config/settings.py (hunk 3 above)
Comment (JdFSilva) [reply labs/config/settings.py:70]: `os.environ.get` always returns a string, so instead of the boolean `True` or `False` it would be a string `"True"` or `"False"`

File: labs/litellm_service/local.py (hunk 4 above)
Comment (JdFSilva) [reply labs/litellm_service/local.py:6]: True, that should come from an env var, I'll add it

File: labs/litellm_service/local.py (hunk 5 above)
Comment (JdFSilva) [reply labs/litellm_service/local.py:23]: because it's also returned at the end, and for now we're just going to use llama3.2
later on we can probably move it to a function argument or an env variable

File: labs/litellm_service/request.py (hunk 6 above)
Comment (JdFSilva) [reply labs/litellm_service/request.py:54]: This was deprecated and no one had removed it yet

**Review by kenvontucky — APPROVED** [review]
